apps/wallet/src/ui/app/components/ledger/SuiLedgerClientProvider.tsx

```diff
@@ -47,23 +51,53 @@ export function SuiLedgerClientProvider({
         return () => suiLedgerClient?.transport.off('disconnect', onDisconnect);
     }, [suiLedgerClient?.transport]);
 
+    const initializeLedgerSignerInstance = useCallback(
+        async (derivationPath: string) => {
+            if (!suiLedgerClient) {
+                try {
+                    const transport = await forceConnectionToLedger();
+                    const newClient = new SuiLedgerClient(transport);
+                    setSuiLedgerClient(newClient);
+
+                    return new LedgerSigner(
+                        newClient,
+                        derivationPath,
+                        api.instance.fullNode
+                    );
+                } catch (error) {
+                    throw new Error('Failed to connect');
```

Comment:
I don't really like how this error handing here and down below is setup atm, I'll go through fix this as a fast follow